client/my-sites/themes/controller.jsx

```diff
@@ -114,8 +114,14 @@ export function redirectToThemeDetails( { res, params: { site, theme, section }
 	res.redirect( '/theme/' + [ theme, redirectedSection, site ].filter( Boolean ).join( '/' ) );
 }
 
-export function redirectTiers( { res, originalUrl }, next ) {
-	const redirectUrl = originalUrl.replace( /\/(free|premium|type)/g, '' );
+export function redirectTiers( { res, originalUrl, params: { tier } }, next ) {
+	if ( tier === undefined ) {
+		return next();
+	}
+	const typeTierRegex = new RegExp( `/type/${ tier }$`, 'g' );
+	const tierRegex = new RegExp( `/${ tier }(/|$)`, 'g' );
+
+	const redirectUrl = originalUrl.replace( typeTierRegex, '' ).replace( tierRegex, '' );
 	if ( redirectUrl === originalUrl ) {
```

Comment:
I think this will never happen, will it? If we reach this point, it mean the URL has a `tier`, and if it has a tier, then it will get removed by this point, and this condition will never be `true`, as the `redirectUrl` will not contain the `tier` (at least) and will be different from the `originalUrl`.

Reply:
You are right, it will never happen. Fixing it!